daemon/loop.go

```diff
@@ -70,11 +75,89 @@ func (d *Daemon) askForSync() {
 }
 
 func (d *Daemon) pullAndSync(logger log.Logger) {
+	started := time.Now().UTC()
+
+	// Pull for new commits
 	if err := d.Checkout.Pull(); err != nil {
 		logger.Log("err", err)
+		return
+	}
+
+	// checkout a working clone so we can mess around with tags later
+	working, err := d.Checkout.WorkingClone()
+	if err != nil {
+		logger.Log("err", err)
+		return
 	}
+	defer working.Clean()
+
+	// update notes and emit events for applied commits
+	revisions, err := working.RevisionsBetween(working.SyncTag+"~1", "HEAD")
+	if isUnknownRevision(err) {
+		// No sync tag, grab all revisions
+		revisions, err = working.RevisionsBefore("HEAD")
+	}
+	if err != nil {
+		logger.Log("err", err)
+	}
+
+	// TODO logging, metrics?
+	// Get a map of all resources defined in the repo
+	allResources, err := d.Cluster.LoadManifests(working.ManifestDir())
+	if err != nil {
+		logger.Log("err", errors.Wrap(err, "loading resources from repo"))
+		return
+	}
+
 	// TODO supply deletes argument from somewhere (command-line?)
-	if err := sync.Sync(d.Checkout.ManifestDir(), d.Cluster, false); err != nil {
+	if err := sync.Sync(allResources, d.Cluster, false); err != nil {
```

Comment:
Yes, we could, but this way if you change the kubernetes cluster "live", and update the git repo it will "fix everything"